### PR TITLE
Catch security exception in get environment call

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

This PR is to fix https://github.com/deepjavalibrary/djl/discussions/1848

The customer run into the get environment security issue. This PR put the environment call into a utility function and catch the exception.

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

**Review by frankfliu — APPROVED** [review]

File: api/src/main/java/ai/djl/util/Utils.java [hunk 1]
```diff
@@ -377,4 +382,35 @@ public static Path getNestedModelDir(Path modelDir) {
...
+     * @param name the name of the environment variable
+     * @return the string value of the variable, or {@code null} if the variable is not defined in
+     *     the system environment or security manager doesn't allow access to the environment
+     *     variable
+     */
+    public static String getenv(String name) {
+        try {
+            return System.getenv(name);
+        } catch (SecurityException e) {
+            logger.error("Security manager doesn't allow access to the environment variable");
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/util/Utils.java:398]:
```suggestion
            logger.warn("Security manager doesn't allow access to the environment variable");
```

File: api/src/main/java/ai/djl/util/Utils.java [hunk 2]
```diff
@@ -377,4 +382,35 @@ public static Path getNestedModelDir(Path modelDir) {
...
+
+    /**
+     * Gets the value of the specified environment variable.
+     *
+     * @param name the name of the environment variable
+     * @return the string value of the variable, or {@code null} if the variable is not defined in
+     *     the system environment or security manager doesn't allow access to the environment
+     *     variable
+     */
+    public static String getenv(String name) {
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/util/Utils.java:394]: I think it's better have an overload function:
```
public static String getenv(String name, String def) {
```

**Review by frankfliu — COMMENTED** [review]
You can merge this PR create a follow up PR to improve the API.

File: api/src/main/java/ai/djl/util/Utils.java (hunk 1 above)
Comment (xyang16) [reply api/src/main/java/ai/djl/util/Utils.java:398]: Changed.

File: api/src/main/java/ai/djl/util/Utils.java (hunk 2 above)
Comment (xyang16) [reply api/src/main/java/ai/djl/util/Utils.java:394]: Added

File: api/src/main/java/ai/djl/util/Utils.java [hunk 3]
```diff
@@ -377,4 +382,54 @@ public static Path getNestedModelDir(Path modelDir) {
...
+        try {
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/util/Utils.java:395]: return getenv(name, null);

File: api/src/main/java/ai/djl/util/Utils.java [hunk 4]
```diff
@@ -377,4 +382,49 @@ public static Path getNestedModelDir(Path modelDir) {
...
+     *
+     * @return the environment as a map of variable names to values
+     */
+    public static Map<String, String> getenv() {
+        try {
+            return System.getenv();
+        } catch (SecurityException e) {
+            logger.warn("Security manager doesn't allow access to the environment variable");
+        }
+        return Collections.unmodifiableMap(new HashMap<>());
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/util/Utils.java:428]:
```suggestion
        return Collections.emptyMap();
```

File: api/src/main/java/ai/djl/util/Utils.java [hunk 5]
```diff
@@ -377,4 +382,49 @@ public static Path getNestedModelDir(Path modelDir) {
         }
         return modelDir.toAbsolutePath();
     }
+
+    /**
+     * Gets the value of the specified environment variable.
+     *
+     * @param name the name of the environment variable
+     * @param def a default value
+     * @return the string value of the variable, or {@code null} if the variable is not defined in
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/util/Utils.java:391]:
```suggestion
     * @return the string value of the variable, or {@code def} if the variable is not defined in
```

File: api/src/main/java/ai/djl/util/Utils.java [hunk 6]
```diff
@@ -30,7 +31,9 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
+import java.util.HashMap;
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/util/Utils.java:34]:
```suggestion
```

Comment (xyang16) [reply api/src/main/java/ai/djl/util/Utils.java]: Done

File: api/src/main/java/ai/djl/util/Utils.java (hunk 5 above)
Comment (xyang16) [reply api/src/main/java/ai/djl/util/Utils.java]: Done

File: api/src/main/java/ai/djl/util/Utils.java (hunk 4 above)
Comment (xyang16) [reply api/src/main/java/ai/djl/util/Utils.java]: Done

File: api/src/main/java/ai/djl/util/Utils.java (hunk 3 above)
Comment (xyang16) [reply api/src/main/java/ai/djl/util/Utils.java:395]: Done

**Pull request merged**